### PR TITLE
[Feature] Support pathStyleAccess for broker load

### LINKED ISSUE #14479: Support pathStyleAccess for broker load
## Enhancement

image

```
Caused by: java.net.UnknownHostException: bucket1.host.testcontainers.internal: Name or service not known
	at [java.net](http://java.net/).Inet4AddressImpl.lookupAllHostAddr(Native Method)
	at [java.net](http://java.net/).InetAddress$2.lookupAllHostAddr(InetAddress.java:866)
	at [java.net](http://java.net/).InetAddress.getAddressesFromNameService(InetAddress.java:1288)
	at [java.net](http://java.net/).InetAddress$NameServiceAddresses.get(InetAddress.java:815)
```

### PR DESCRIPTION
**Pull request by jiacheng-celonis**
## What type of PR is this：
- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #14479

## Problem Summary(Required) ：

Add a new broker config option `"fs.s3a.path.style.access"` to allow toggling the path style. The option value must either be "true" or "false". Default value is "false".

```
WITH BROKER "broker1"
(
  "fs.s3a.access.key" = "myaccesskey",
  "fs.s3a.secret.key" = "mysecretkey",
  "fs.s3a.endpoint" = "host.testcontainers.internal:49651"
  "fs.s3a.path.style.access" = "true"
)
```
## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3
  - [ ] 2.2

### CODE REVIEW

File: fs_brokers/apache_hdfs_broker/src/main/java/com/starrocks/broker/hdfs/FileSystemManager.java [hunk 1]
```diff
@@ -450,6 +451,7 @@ public BrokerFileSystem getS3AFileSystem(String path, Map<String, String> proper
         String accessKey = properties.getOrDefault(FS_S3A_ACCESS_KEY, "");
         String secretKey = properties.getOrDefault(FS_S3A_SECRET_KEY, "");
         String endpoint = properties.getOrDefault(FS_S3A_ENDPOINT, "");
+        String pathStyleAccess = properties.getOrDefault(FS_S3A_PATH_STYLE_ACCESS, "true");
```
Comment (rcauble) [inline fs_brokers/apache_hdfs_broker/src/main/java/com/starrocks/broker/hdfs/FileSystemManager.java:454]: Default needs to be "false" for backwards compatibility.

**Review by rcauble — DISMISSED** [review]

File: fs_brokers/apache_hdfs_broker/src/main/java/com/starrocks/broker/hdfs/FileSystemManager.java (hunk 1 above)
Comment (jiacheng-celonis) [reply fs_brokers/apache_hdfs_broker/src/main/java/com/starrocks/broker/hdfs/FileSystemManager.java]: done

**Review by dengliu — APPROVED** [review]

**Review by chaoyli — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Review by imay — APPROVED** [review]

**Comment by chaoyli** [comment]
@mergify backport branch-2.2 branch-2.3 branch-2.4 branch-2.5

**Pull request merged**